tests/integration/Core/Image/ImageStorage/ImageStorageTest.php

```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * @dataProvider providerOfFieldData
+     *
+     * @depends testStoreFieldDataDuringCreate
+     */
+    public function testStoreFieldDataDuringUpdate(VersionInfo $versionInfo, Field $field): void
+    {
+        $binaryFile = $this->runCommonStoreFieldDataMocks($field);
+
+        $this->redecorator
+            ->expects(self::exactly(2))
+            ->method('redecorateFromSource')
+            ->with($binaryFile->uri)
+            ->willReturn($binaryFile->uri);
+
+        $this->storage->storeFieldData($versionInfo, $field, $this->getContext());
+
+        self::assertEquals(1, $this->gateway->countImageReferences($binaryFile->uri));
+    }
+
+    private function runCommonStoreFieldDataMocks(Field $field): BinaryFile
+    {
+        $this->filePathNormalizer
+            ->expects(self::once())
+            ->method('normalizePath')
+            ->willReturn($targetPath = '1/8/6/232-eng-GB/' . $field->value->externalData['fileName']);
+
+        $this->ioService
+            ->expects(self::once())
+            ->method('newBinaryCreateStructFromLocalFile')
+            ->with($field->value->externalData['inputUri'])
+            ->willReturn($newBinaryFileCreateStruct = new BinaryFileCreateStruct());
+
+        $this->ioService
+            ->expects(self::once())
+            ->method('createBinaryFile')
+            ->with($newBinaryFileCreateStruct)
+            ->willReturn($binaryFile = new BinaryFile(
+                [
+                    'id' => $targetPath,
+                    'uri' => $targetPath,
+                ]
+            ));
+
+        $this->ioService
+            ->expects(self::once())
+            ->method('getMimeType')
+            ->with($binaryFile->id)
+            ->willReturn('image/jpeg');
+
+        return $binaryFile;
+    }
+
+    public function providerOfFieldData(): array
+    {
+        $path = __DIR__ . '/image.jpg';
+
+        $field = new Field();
+        $field->id = 125;
+        $field->fieldDefinitionId = 232;
+        $field->type = 'ezimage';
+        $field->versionNo = 1;
+        $field->value = new FieldValue([
+            'externalData' => [
+                'id' => null,
+                'path' => $path,
+                'inputUri' => $path,
+                'fileName' => 'image.jpg',
+                'fileSize' => '12345',
+                'mimeType' => 'image/jpeg',
+                'width' => null,
+                'height' => null,
+                'alternativeText' => null,
+                'imageId' => null,
+                'uri' => null,
+                'additionalData' => [],
+            ],
+        ]);
+
+        $versionInfo = new VersionInfo([
+            'contentInfo' => new ContentInfo([
+                'id' => 236,
+                'contentTypeId' => 25,
+            ]),
+            'versionNo' => 1,
+        ]);
+
+        return [
+            [$versionInfo, $field],
+        ];
```

Comment:
new fashion continuation:
```suggestion
        yield [$versionInfo, $field];
```
;-)